toolchain/check/function.cpp

```diff
@@ -315,7 +315,7 @@ auto MergeFunctionRedecl(Context& context, SemIR::LocId loc_id,
     prev_function.implicit_param_refs_id = new_function.implicit_param_refs_id;
     prev_function.param_refs_id = new_function.param_refs_id;
     prev_function.return_type_id = new_function.return_type_id;
-    prev_function.return_slot_id = new_function.return_slot_id;
+    prev_function.return_storage_id = new_function.return_storage_id;
```

Comment:
Would it be worth asserting new_function.return_slot is NotComputed? (tieing in with my other comment that maybe it shouldn't always be NotComputed)

Reply:
Changed to merge the slot info.